test/system/system_suite_test.go

```diff
@@ -29,17 +28,63 @@ func TestSystem(t *testing.T) {
 
 var _ = SynchronizedBeforeSuite(func() {
 	//this runs once for the whole suite
+	worker = &destination{
+		context:       getEnvOrDefault("WORKER_CONTEXT", "kind-worker"),
+		checkExitCode: true,
+		name:          getEnvOrDefault("WORKER_NAME", "worker-1"),
+	}
+	platform = &destination{
+		context:       getEnvOrDefault("PLATFORM_CONTEXT", "kind-platform"),
+		checkExitCode: true,
+		name:          getEnvOrDefault("PLATFORM_NAME", "platform-cluster"),
+	}
+
+	if getEnvOrDefault("PLATFORM_SKIP_SETUP", "false") == "true" {
+		return
+	}
+
 	var err error
 	testTempDir, err = os.MkdirTemp(os.TempDir(), "systest")
 	Expect(err).NotTo(HaveOccurred())
 	initK8sClient()
 	tmpDir, err := os.MkdirTemp(os.TempDir(), "systest")
 	Expect(err).NotTo(HaveOccurred())
+
 	platform.kubectl("apply", "-f", "../../hack/destination/gitops-tk-install.yaml")
+	platform.kubectl("apply", "-f", "./assets/bash-promise/deployment.yaml")
 	platform.kubectl("apply", "-f", catAndReplaceFluxResources(tmpDir, "./assets/git/platform_gitops-tk-resources.yaml"))
 	platform.kubectl("apply", "-f", catAndReplaceFluxResources(tmpDir, "./assets/git/platform_kratix_destination.yaml"))
 	os.RemoveAll(tmpDir)
-}, func() {})
+}, func() {
+	//this runs before each test
+
+	//This variable gets set in func above, but only for 1 of the nodes, so we set
+	//it again here to ensure all nodes have it
+	worker = &destination{
+		context:       getEnvOrDefault("WORKER_CONTEXT", "kind-worker"),
+		checkExitCode: true,
+		name:          getEnvOrDefault("WORKER_NAME", "worker-1"),
+	}
+	platform = &destination{
+		context:       getEnvOrDefault("PLATFORM_CONTEXT", "kind-platform"),
+		checkExitCode: true,
+		name:          getEnvOrDefault("PLATFORM_NAME", "platform-cluster"),
+	}
+
+	endpoint = getEnvOrDefault("BUCKET_ENDPOINT)", "localhost:31337")
+	secretAccessKey = getEnvOrDefault("BUCKET_SECRET_KEY", "minioadmin")
+	accessKeyID = getEnvOrDefault("BUCKET_ACCESS_KEY", "minioadmin")
+	useSSL = getEnvOrDefault("BUCKET_SSL", "false") == "true"
```

Comment:
```suggestion
	useSSL = os.GetEnv("BUCKET_SSL") == "true"
```

Reply:
this is the most derik comment ever

Reply:
its called "writing good code" @aclevername